catroid/src/main/java/org/catrobat/catroid/copypaste/ClipboardHandler.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.copypaste;
+
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.io.storage.DirectoryPathInfo;
+
+import java.io.IOException;
+
+public class ClipboardHandler {
+
+	public static final String TAG = ClipboardHandler.class.getSimpleName();
+
+	private static ClipboardHandler instance;
+
+	private Clipboard clipboard;
+
+	private static final DirectoryPathInfo CLIPBOARD_DIRECTORY = new DirectoryPathInfo(new DirectoryPathInfo(Constants.DEFAULT_ROOT), "clipboard");
+
+	public static ClipboardHandler getInstance() {
```

Comment:
well, we need the member clipboard to be persistent (through different activities). I'm going to look into it.